parlai/chat_service/services/websocket/sockets.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from tornado.websocket import WebSocketHandler
+import uuid
+import logging
+
+
+def get_rand_id():
+    return str(uuid.uuid4())
+
+
+class MessageSocketHandler(WebSocketHandler):
+    subs = {}
+
+    def __init__(self, *args, **kwargs):
+        self.message_callback = kwargs.pop('message_callback', None)
+        self.sid = get_rand_id()
+        super(MessageSocketHandler, self).__init__(*args, **kwargs)
+
+    def open(self):
+        """
+        Opens a websocket and assigns a random UUID that is stored in the class-level
+        `subs` variable
+        """
+        if self.sid not in self.subs.values():
+            self.subs[self.sid] = self
+            logging.info(f"Opened new socket from ip: {self.request.remote_ip}")
+            logging.info(f"Current subscribers: {self.subs}")
+
+    def on_close(self):
+        """Runs when a socket is closed"""
+        del self.subs[self.sid]
+
+    def on_message(self, message):
+        """Callback that runs when a new message is received from a client"""
+        logging.info('websocket message from client: {}'.format(message))
+        self.message_callback(message, self.sid)
```

Comment:
will this fail if there is no `self.message_callback`? should we provide a default one?

Reply:
Defined a default that outputs a warning